src/caffe/util/gpu_memory.cpp

```diff
@@ -37,7 +37,7 @@ void GPUMemory::Manager::init(const vector<int>& gpus, Mode m, bool debug) {
       // Just in case someone installed 'no cleanup' arena before
       delete cub_allocator_;
       cub_allocator_ = new cub::CachingDeviceAllocator(BIN_GROWTH, MIN_BIN,
-          MAX_BIN, MAX_CACHED_BYTES, false, debug_);
+          MAX_BIN, MAX_CACHED_BYTES, true, debug_);
```

Comment:
On a second note, since the only instance of mgr is global, having cleanup is useless and could only add latency and more opportunities for deadlocks.